File: mailblastr-java/src/main/java/com/mailblastr/resources/Logs.java
package com.mailblastr.resources;

import com.mailblastr.MailblastrResponse;
import com.mailblastr.http.ApiClient;
import com.mailblastr.http.Query;
import com.mailblastr.requests.ListLogsParams;

/** API request logs — {@code mailblastr.logs()} (read-only). */
public final class Logs extends Resource {
    public Logs(ApiClient api) { super(api); }

    /** {@code GET /logs} */
    public MailblastrResponse list() { return list(null); }

    /**
     * List API request logs. Cursor-paginated with optional server-side
     * {@code method} / {@code status} filters.
     */
    public MailblastrResponse list(ListLogsParams params) {
        Query q = new Query();
        if (params != null) {
            q.add("limit", params.getLimit())
             .add("after", params.getAfter())
             .add("before", params.getBefore())
             .add("method", params.getMethod())
             .add("status", params.getStatus());
        }
        return api.request("GET", "/logs" + q);
    }

    /** Retrieve one log entry (includes request/response bodies). {@code GET /logs/:id} */
    public MailblastrResponse get(String id) {
        return api.request("GET", "/logs/" + enc(id));
    }
}

File: mailblastr-php/src/Resources/Logs.php
<?php

declare(strict_types=1);

namespace Mailblastr\Resources;

use Mailblastr\Client;

/**
 * API request logs (read-only).
 */
class Logs extends Resource
{
    /**
     * List API request logs. Cursor-paginated with optional server-side
     * filters. GET /logs
     *
     * @param array $params limit, after, before, method (exact HTTP method,
     *                      e.g. 'POST'), status (exact response status, e.g. 429).
     */
    public function list(array $params = []): array
    {
        return $this->client->request('GET', '/logs' . $this->paginationQuery($params, ['method', 'status']));
    }

    /** Retrieve a log entry (includes request/response bodies). GET /logs/:id */
    public function get(string $id): array
    {
        return $this->client->request('GET', '/logs/' . Client::e($id));
    }
}
